### PR TITLE
update angular2

### PR DESCRIPTION
**Pull request by folliked**
- add webpack
- remove jspm and systemjs
- update angularJs 2 in stable version  

it remains minifying the files,  cleaning and tweaking config webpack

### CODE REVIEW

File: karma.conf.js [hunk 1]
```diff
@@ -0,0 +1 @@
+module.exports = require('./config/karma.conf.js');
```
Comment (Nek-) [inline karma.conf.js]: (missing new line at EOF)

File: src-js/app/user/session.service.ts [hunk 2]
```diff
@@ -22,14 +22,18 @@ export class Session {
     init() : Promise<Session> {
 
         if (this.active !== null) {
-            return new Promise<Session>((resolve, reject) => { resolve(this); });
+            return new Promise<Session>((resolve: any, reject: any) => {
+                resolve(this);
+                reject('nope');
```
Comment (Nek-) [inline src-js/app/user/session.service.ts]: Why ???

File: src-js/app/user/session.service.ts [hunk 3]
```diff
@@ -22,14 +22,18 @@ export class Session {
...
+            });
         }
 
-        return new Promise<Session>((resolve, reject) => {
+        return new Promise<Session>((resolve: any, reject: any) => {
             this.renew().subscribe((res:Response) => {
                 this.active = res.status === 200;
                 resolve(this);
             }, reject);
+            reject('nope');
```
Comment (Nek-) [inline src-js/app/user/session.service.ts]: Why ???

File: src-js/app/user/session.service.ts [hunk 4]
```diff
@@ -41,19 +45,20 @@ export class Session {
...
     }
 
     logout() {
-        return new Promise<Session>((resolve, reject) => {
+        return new Promise<Session>((resolve: any, reject: any) => {
             this.http.get('/back/logout').subscribe((res:Response) => {
                 this.active = false;
                 resolve(this);
             }, reject);
+            reject('nope');
```
Comment (Nek-) [inline src-js/app/user/session.service.ts]: Why ???

File: src-js/gulp/gulp-connect-symfony.js [hunk 5]
```diff
@@ -1,145 +0,0 @@
-'use strict';
-
-var fs = require('fs');
-var binVersionCheck = require('bin-version-check');
-var http = require('http');
-var open = require('opn');
-var spawn = require('child_process').spawn;
-var exec = require('child_process').exec;
-
-module.exports = (function () {
```
Comment (Nek-) [inline src-js/gulp/gulp-connect-symfony.js]: so much work/experimentation to finally drop it without any regret... :')

File: src/CiTron/Controller/DefaultController.php [hunk 6]
```diff
@@ -31,8 +31,9 @@ public function frontAction(Request $request)
         }
 
         if (in_array($this->container->getParameter('kernel.environment'), ['dev', 'test'])) {
-            return new Response(file_get_contents($rootDir . '/../src-js/dev.html'));
+            return new Response(file_get_contents($rootDir . '/../src-js/index.html'));
         }
+
         if ($this->container->getParameter('kernel.environment') === 'prod') {
             return new Response(file_get_contents($rootDir . '/../src-js/prod.html'));
```
Comment (Nek-) [inline src/CiTron/Controller/DefaultController.php]: I guess this is not accurate anymore. Please put theses 3 lines in comments and add a comment that says why we did that.

File: webpack.config.js [hunk 7]
```diff
@@ -0,0 +1 @@
+module.exports = require('./config/webpack.dev.js');
```
Comment (Nek-) [inline webpack.config.js]: missing new line at EOF

File: src-js/app/app.component.ts [hunk 8]
```diff
@@ -57,7 +55,9 @@ export class AppComponent implements OnInit {
     ngOnInit() {
         this.session.init().then(
             (session:Session) => this.logged = session.isActive(),
-            (error) => console.error(error)
+            // (reject: any) => console.log("then: " + reject)
```
Comment (folliked) [inline src-js/app/app.component.ts]: ok i will do

File: karma.conf.js (hunk 1 above)
Comment (folliked) [inline karma.conf.js]: ok i will do

File: webpack.config.js (hunk 7 above)
Comment (folliked) [inline webpack.config.js]: ok i will do

File: src-js/app/style/global.less [hunk 9]
```diff
@@ -1,4 +1,4 @@
-@import "bootstrap";
+@import "~bootstrap/less/bootstrap";
```
Comment (folliked) [inline src-js/app/style/global.less]: explanation here : https://github.com/webpack/less-loader#imports

Comment (Nek-) [inline src-js/app/style/global.less]: awesome

File: src-js/gulp/gulp-connect-symfony.js (hunk 5 above)
Comment (valanz) [inline src-js/gulp/gulp-connect-symfony.js]: @Nek- :smile:

**Pull request merged**